packages/react-component-library/src/primitives/FloatingBox/FloatingBox.tsx

```diff
@@ -89,11 +91,11 @@ export const FloatingBox: React.FC<FloatingBoxProps> = ({
           {renderTarget}
         </StyledTarget>
       )}
-      <Transition in={isVisible} timeout={0} unmountOnExit>
+      <Transition nodeRef={nodeRef} in={isVisible} timeout={0} unmountOnExit>
         {(state) => (
           <StyledFloatingBox
             style={{ ...styles.popper, ...TRANSITION_STYLES[state] }}
-            ref={floatingElementRef}
+            ref={mergeRefs([nodeRef, floatingElementRef])}
```

Comment:
Local `nodeRef` merged because `Transition` doesn't like the special stateful refs `useFloatingBox` utilises.